website/addons/wiki/views.py

```diff
@@ -245,14 +219,45 @@ def project_wiki_delete(auth, wname, **kwargs):
 
 
 @must_be_valid_project  # returns project
-@must_have_permission('write')  # returns user, project
-@must_not_be_registration
+@must_be_contributor_or_public
 @must_have_addon('wiki', 'node')
-def project_wiki_edit(auth, wname, **kwargs):
+def project_wiki_view(auth, wname, path=None, **kwargs):
     node = kwargs['node'] or kwargs['project']
-    wiki_name = wname.strip()
+    anonymous = has_anonymous_link(node, auth)
+    wiki_name = (wname or '').strip()
     wiki_key = to_mongo_key(wiki_name)
     wiki_page = node.get_wiki_page(wiki_name)
+    toc = _serialize_wiki_toc(node, auth=auth)
+    can_edit = node.has_permission(auth.user, 'write') and not node.is_registration
+    versions = _get_wiki_versions(node, wiki_name, anonymous=anonymous)
+
+    # Parse URL to determine default panel display
+    view_settings = {}
+    param_list = path.lower().split('/') if path else []
```

Comment:
Okay, @sloria and I talked. We've decided to simplify the URL structure quite a bit and get rid of the implicit states. The main URL that you go to will be the wiki view url `/<pid>/wiki/<wid>/`. We'll redirect from  `/edit/` and `/history/` to the new url structure. 

For everything else, we'll use query parameters/ `?compare&view&edit&menu`. If there's anything at all in query parameters, then view has to be there otherwise it means the column is disabled. Also, compare and view can have optional parameters to indicate version, so `?compare=5&view=preview`, for example.

I think we'll still have implicit versions for compare that are situational, or we need a special `?compare=-1` option (or something more verbose) to indicate that we want the most recent non-current version. Otherwise, by default view is current unless edit is on at the beginning in which case it's preview, and compare by default is most recent (i.e. -1) unless edit is on and view set to preview in which case it's current. (This may need to be simplified before everything is over).